crypto/x509/x_crl.c

```diff
@@ -383,8 +383,11 @@ static int def_crl_lookup(X509_CRL *crl,
                           X509_NAME *issuer)
 {
     X509_REVOKED rtmp, *rev;
-    int idx;
-    rtmp.serialNumber = *serial;
+    int idx, num;
+
+    if ((num = sk_X509_REVOKED_num(crl->crl.revoked)) < 0)
```

Comment:
Well, the initial idea was to kind of make it as close as possible to original. So I actually started with `crl->crl.revoked == NULL`, then saw possibility to pull num early, but equivalent check would be <0, because num being zero is permitted state. But know what? On second though I'd say that `crl->crl.revoked == NULL` is actually more readable. In sense that it asks for less questions... I'd also spotted couple of nits... Pushed commit.

Reply:
Makes sense.  Reconfirm.